leku/src/main/java/com/schibstedspain/leku/search/LekuSearchFragment.java

```diff
@@ -0,0 +1,248 @@
+import com.schibstedspain.leku.NullView;
+import com.schibstedspain.leku.R;
+import java.util.List;
+
+public class LekuSearchFragment extends Fragment {
+
+  private LekuSearchCallback lekuSearchCallback;
+
+  public static LekuSearchFragment newInstance() {
+    return new LekuSearchFragment();
+  }
+
+  private static final String LAST_LOCATION_QUERY = "last_location_query";
+  private static final int MIN_CHARACTERS = 2;
+  private static final int REQUEST_PLACE_PICKER = 6655;
+  private static final int CONNECTION_FAILURE_RESOLUTION_REQUEST = 9000;
+
+  private Toolbar toolbar;
+  private EditText searchView;
+  private ImageView clearSearchButton;
+  private TextWatcher textWatcher;
+  private MenuItem searchOption;
+
+  @Nullable
+  @Override
+  public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+    super.onCreateView(inflater, container, savedInstanceState);
+    return inflater.inflate(R.layout.toolbar_search, null, false);
+  }
+
+  @Override
+  public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
+    super.onViewCreated(view, savedInstanceState);
+
+    bindViews(view);
+    setupToolbar();
+    setUpClearListener();
+    setUpSearchViewListeners();
+  }
+
+  private void bindViews(View view) {
+    toolbar = (Toolbar) view.findViewById(R.id.map_search_toolbar);
+    searchView = (EditText) view.findViewById(R.id.leku_search);
+    clearSearchButton = (ImageView) view.findViewById(R.id.leku_clear_search_image);
+  }
+
+  private void setupToolbar() {
+    toolbar.inflateMenu(R.menu.toolbar_menu);
+    searchOption = toolbar.getMenu().findItem(R.id.action_voice);
+    toolbar.setOnMenuItemClickListener(this::onToolbarMenuItemSelected);
+  }
+
+  private boolean onToolbarMenuItemSelected(MenuItem item) {
+    if (item.getItemId() == R.id.action_voice) {
+      onVoiceItemSelected();
+      return true;
+    }
+    return false;
+  }
+
+  public void clear() {
+    searchView.setText("");
+    closeKeyboard();
+  }
+
+  private void onVoiceItemSelected() {
+    if (searchView.getText().toString().isEmpty()) {
+      startVoiceRecognitionActivity();
+    } else {
+      retrieveLocationFrom(searchView.getText().toString());
+      closeKeyboard();
+    }
+  }
+
+  private void startVoiceRecognitionActivity() {
+    Intent intent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);
+    intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL, RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);
+    intent.putExtra(RecognizerIntent.EXTRA_PROMPT, getString(R.string.voice_search_promp));
+    intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE, getString(R.string.voice_search_extra_language));
+
+    if (checkPlayServices()) {
+      try {
+        startActivityForResult(intent, REQUEST_PLACE_PICKER);
+      } catch (ActivityNotFoundException e) {
+        Log.d(LocationPickerActivity.class.getName(), e.getMessage());
```

Comment:
Please don't use Log. Probably this line is outdated with the latest merged PR that replaced the Log with a track method